### ISSUE TITLE
[HOLD for payment 2024-12-07] [$250] Workspace - Currency is shown empty for a workspace created in OD

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#52190]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.59-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+shsb22co12112@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Prerequisite
Make sure you are signed in the same account with OD and ND
1. Navigate to OD
2. Create a new workspace (Can be either collect or control)
3. Navigate to ND and open the workspace settings of the workspace that was created in OD
4. Observe the Default currency in the profile page

## Expected Result:
The default currency is dispalyed

## Actual Result:
The default currency is not displayed, "-" is displayed instead

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/f3ee8109-7733-4153-a05b-81ea03ef9a1a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021854617872012775593</li>
        <li>Upwork Job ID: 1854617872012775593</li>
        <li>Last Price Increase: 2024-11-21</li>
<li>Automatic offers: </li>
<ul>
<li>Tony-MK | Contributor | 105019925</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @lschurr / @zanyrenney</details>

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Currency is shown empty for a workspace created in OD.

### What is the root cause of that problem?

The `policy?.outputCurrency` is not found, hence the `outputCurrency` will be an empty string. 

https://github.com/Expensify/App/blob/2ef91b8b9d36f9534969d7bb0e3133b2a848e7a4/src/pages/workspace/WorkspaceProfilePage.tsx#L56-L58

As a result, the `formattedCurrency` will be displayed as a dash.

### What changes do you think we should make in order to solve the problem?


Since there are ways of finding the `outputCurrency`, let's get it by finding it from the personal policy or the default mileage rate currency which are equal to the `policy?.outputCurrency`. 

```
const outputCurrency = policy?.outputCurrency ??
                       DistanceRequestUtils.getDefaultMileageRate(policy).currency ??
                       PolicyUtils.getPersonalPolicy()?.outputCurrency ?? ''
```

In case `policy?.outputCurrency` is not found, like in this case, we created the same condition, as above, in other places such as [`IOURequestStepDistance`](https://github.com/Expensify/App/blob/2ef91b8b9d36f9534969d7bb0e3133b2a848e7a4/src/pages/iou/request/step/IOURequestStepDistance.tsx#L136) and [`TransactionUtils.calculateAmountForUpdatedWaypointOrRate`]([src/libs/TransactionUtils/index.ts](https://github.com/Expensify/App/blob/20404082dc199e735d711d420e0e47436dd8609f/src/libs/TransactionUtils/index.ts#L884)).

**Comment by mohit6789** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Currency is showing empty when workspace created in OD.

### What is the root cause of that problem?
Currency is showing empty because `outputCurrency` does not return from the `policy` object. Because of that `formattedCurrency` is displayed as '-'

```
const outputCurrency = policy?.outputCurrency ?? '';
const currencySymbol = currencyList?.[outputCurrency]?.symbol ?? '';
const formattedCurrency = !isEmptyObject(policy) && !isEmptyObject(currencyList) ? `${outputCurrency} - ${currencySymbol}` : '';
```

### What changes do you think we should make in order to solve the problem?
To fix issue we need to make changes in two places. 

1. In `WorkspaceProfilePage.tsx` Replace `const outputCurrency = policy?.outputCurrency ?? '';` with `const outputCurrency = policy?.outputCurrency ?? PolicyUtils.getPersonalPolicy()?.outputCurrency ?? CONST.CURRENCY.USD;` as we used in other places.
```
const outputCurrency = policy?.outputCurrency ?? PolicyUtils.getPersonalPolicy()?.outputCurrency ?? CONST.CURRENCY.USD;
```
2. When user press on currency it should be selected in currency [sidebar](https://url.upwork.com/_01rL-UyDMzeJ7HgThtf5xlrJKE1ezo-jAH). To fix this issue we need to make changes in  `WorkspaceProfileCurrencyPage.tsx`.
```
const outputCurrency = policy?.outputCurrency ?? PolicyUtils.getPersonalPolicy()?.outputCurrency ?? CONST.CURRENCY.USD;

<CurrencySelectionList
    searchInputLabel={translate('workspace.editor.currencyInputLabel')}
    onSelect={onSelectCurrency}
    initiallySelectedCurrencyCode={outputCurrency}
/>
```
### What alternative solutions did you explore? (Optional)
NA

**Comment by lschurr** [comment]
@Ollyws could you review these proposals?

**Comment by mohit6789** [comment]
@Ollyws still I can see issue https://url.upwork.com/_01oz_yI_oSgfFCFkjS8XfqcDMSf0umFos5 , https://url.upwork.com/_01oz_yI_oSgfHYgTl2VwpR4j1PjDI_KPfb

**Comment by mohit6789** [comment]
@Ollyws I can see same issue in latest main

**Comment by twisterdotcom** [comment]
Relabelling as Lauren is OOO now.

**Comment by mjasikowski** [comment]
@Ollyws any thoughts on @mohit6789's point? Sidebar selection needs to be addressed too, as it currently uses `policy?.outputCurrency`, which won't have a value.

**Comment by mjasikowski** [comment]
@Ollyws I don't agree that fixing sidebar selection is out of scope, as we should be fixing bugs and their consequences as well. Your point about @Tony-MK's solution being first and tackling the gist of the issue is correct though.
Let's go with @Tony-MK's proposal, but please also fix the sidebar.
 
@lschurr as @mohit6789's proposal contains a valid point that builds on @Tony-MK's solution, let's pay $50 to @mohit6789 for his contribution

**Comment by mjasikowski** [comment]
@Tony-MK any updates here?

**Comment by Tony-MK** [comment]
Working on the PR. It should be ready by tomorrow.

**Comment by Tony-MK** [comment]
Hey @Ollyws, the [PR](https://github.com/Expensify/App/pull/53132) is ready for review.

**Comment by lschurr** [comment]
Removing you Zany :)

**Comment by lschurr** [comment]
Payment summary:

- Contributor: @Tony-MK paid $250 in [Upwork](https://www.upwork.com/nx/wm/workroom/39057887/overview)
- Reviewer: @Ollyws please request $250 in NewDot

**Comment by garrettmknight** [comment]
$250 approved for @Ollyws

**Comment by Tony-MK** [comment]
> @lschurr as @mohit6789's proposal contains a valid point that builds on @Tony-MK's solution, let's pay $50 to @mohit6789 for his contribution

Hey @lschurr, I think you skipped this [comment](https://github.com/Expensify/App/issues/52190#issuecomment-2493504637) by @mjasikowski.

**Comment by mohit6789** [comment]
@lschurr I think you missed to pay 50$ to me as mention by @mjasikowski  in  comment https://github.com/Expensify/App/issues/52190#issuecomment-2493504637

**Comment by lschurr** [comment]
Sorry about that! I just sent the offer in Upwork @mohit6789 - https://www.upwork.com/nx/wm/offer/105698823

**Comment by lschurr** [comment]
Done.

**Comment by trjExpensify** [comment]
👋 Hi! Coming from here. Can we please get the checklist for this issue, and the regression test? Thanks!
